File: src/main/java/dk/fitfit/oneliner/service/OneLinerServiceInterface.java
package dk.fitfit.oneliner.service;

import dk.fitfit.oneliner.domain.OneLiner;

import java.util.List;
import java.util.Set;

public interface OneLinerServiceInterface {
	Set<OneLiner> findByAllTags(Set<String> tags);

	List<OneLiner> findAll();

	OneLiner save(OneLiner oneLiner);

	OneLiner findOne(long id);
}

File: src/main/kotlin/dk/fitfit/oneliner/service/OneLinerServiceInterface.kt
package dk.fitfit.oneliner.service

import dk.fitfit.oneliner.domain.OneLiner

interface OneLinerServiceInterface {
    fun findByAllTags(tags: Set<String>): Set<OneLiner>

    fun findAll(): List<OneLiner>

    fun save(oneLiner: OneLiner): OneLiner

    fun getOne(id: Long): OneLiner
}
